Codegen error when operating on sets of a type and of a range of the same type

If you perform an operation on 2 sets with the same base type but one being a range type of that type, you get a codegen error instead of a type mismatch.

### Example
```nim
var x: set[range[1u8..5u8]]
x.incl(1)
x.incl(3)
x.incl(5)
if {1u8, 5} < x:
  echo "successful"
discard {1u8, 3, 5} + x
```

### Current Output
```
/usercode/nimcache/@min.nim.c: In function 'NimMainModule':
/usercode/nimcache/@min.nim.c:180:50: error: cast specifies array type
   T4_ = ((TM__JXJjWlLylejo6nWONmO6cA_2[T3_] & ~ ((tySet_tyUInt8__k3HXouOuhqAKq0dx450lXQ) (x__EGWKGClUlppqAonIP2rgfg))[T3_]) == 0);
                                                  ^
/usercode/nimcache/@min.nim.c:182:58: error: cast specifies array type
 if (T4_) T4_ = (nimCmpMem(TM__JXJjWlLylejo6nWONmO6cA_2, ((tySet_tyUInt8__k3HXouOuhqAKq0dx450lXQ) (x__EGWKGClUlppqAonIP2rgfg)), 32) != 0);
                                                          ^
/usercode/nimcache/@min.nim.c:191:51: error: cast specifies array type
   T8_[T7_] = TM__JXJjWlLylejo6nWONmO6cA_5[T7_] | ((tySet_tyUInt8__k3HXouOuhqAKq0dx450lXQ) (x__EGWKGClUlppqAonIP2rgfg))[T7_];
```

### Expected Output
type mismatch, got set[range[1u8..5u8]] but expected set[uint8] etc

### Additional Information
* Very slightly related: #3540

```
$ nim -v
Nim Compiler Version 1.2.6
```

--- a/tests/sets/t15435.nim
+++ b/tests/sets/t15435.nim
@@ -1,3 +1,4 @@
+# bug #15435
 discard """
 errormsg: "type mismatch: got <set[uint8], set[range 1..5(uint8)]>"
 nimout: '''t15435.nim(28, 13) Error: type mismatch: got <set[uint8], set[range 1..5(uint8)]>
@@ -10,7 +11,6 @@ proc `<`[T](x, y: set[T]): bool
 
 expression: {1'u8, 5} < x'''
 """
-
 
 
 

--- a/tests/sets/t15435.nim
+++ b/tests/sets/t15435.nim
@@ -1,0 +1,29 @@
+discard """
+errormsg: "type mismatch: got <set[uint8], set[range 1..5(uint8)]>"
+nimout: '''t15435.nim(28, 13) Error: type mismatch: got <set[uint8], set[range 1..5(uint8)]>
+but expected one of:
+proc `<`[T](x, y: set[T]): bool
+  first type mismatch at position: 2
+  required type for y: set[T]
+  but expression 'x' is of type: set[range 1..5(uint8)]
+20 other mismatching symbols have been suppressed; compile with --showAllMismatches:on to see them
+
+expression: {1'u8, 5} < x'''
+"""
+
+
+
+
+
+
+
+## line 20
+var
+  x: set[range[1u8..5u8]]
+
+x.incl(1)
+x.incl(3)
+x.incl(5)
+
+if {1u8, 5} < x:
+  echo "successful"

